kod/object/active/holder/nomoveon/battler/player.kod

```diff
@@ -6588,9 +6588,13 @@ messages:
       if report AND iChange > 0
       {
          oRoom = Send(self, @GetOwner);
-         Post(self,@MsgSendUser,#message_rsc=player_improved,#parm1=spellName);
-         Send(oRoom, @SomethingWaveRoom, #what=self, 
-              #wave_rsc=player_improved_wav_rsc);
+         Post(self,@MsgSendUser,#message_rsc=player_improved,#parm1=spellName);   
+         
+         if NOT (isClass(self,&Admin) AND Send(self,@IsHidden))
```

Comment:
The bug says "DM or Admin", so change this to &DM instead?

Reply:
This was written to match the related logic, which returns from `Spell`:`CastSpell` when Admin and hidden. Updating that as well as a part of this.